File: EPI/9.11/swift/9.11_test.swift
import Darwin

var queue = Queue<Int>()
assert(queue.count == 0)

queue.enqueue(10)
assert(queue.count == 1)

queue.enqueue(15)
assert(queue.count == 2)

assert(queue.dequeue() == 10)
assert(queue.count == 1)

queue.enqueue(20)
assert(queue.count == 2)

queue.enqueue(25)
assert(queue.count == 3)

assert(queue.dequeue() == 15) // Dequeued from array[1]
assert(queue.count == 2)

queue.enqueue(30)
assert(queue.count == 3)

queue.enqueue(35)
assert(queue.count == 4)

assert(queue.dequeue() == 20)
assert(queue.dequeue() == 25)
assert(queue.dequeue() == 30)
assert(queue.dequeue() == 35)
assert(queue.count == 0)

File: EPI/9.11/typescript/9.11_test.ts
import Queue from './9.11';

declare function require(name: string): any;
const assert = require('assert');

const queue = new Queue<number>()
assert.strictEqual(queue.length, 0);

queue.enqueue(10);
assert.strictEqual(queue.length, 1);

queue.enqueue(15);
assert.strictEqual(queue.length, 2);

assert.strictEqual(queue.dequeue(), 10)
assert.strictEqual(queue.length, 1);

queue.enqueue(20);
assert.strictEqual(queue.length, 2);

queue.enqueue(25);
assert.strictEqual(queue.length, 3);

assert.strictEqual(queue.dequeue(), 15) // Dequeued from array[1]
assert.strictEqual(queue.length, 2);

queue.enqueue(30);
assert.strictEqual(queue.length, 3);

queue.enqueue(35);
assert.strictEqual(queue.length, 4);

assert.strictEqual(queue.dequeue(), 20)
assert.strictEqual(queue.dequeue(), 25)
assert.strictEqual(queue.dequeue(), 30)
assert.strictEqual(queue.dequeue(), 35)
assert.strictEqual(queue.length, 0);
